subprojects/docs/src/samples/codeQuality/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,69 @@
+// tag::use-checkstyle-plugin[]
+// tag::use-codenarc-plugin[]
+// tag::use-findbugs-plugin[]
+// tag::use-jdepend-plugin[]
+// tag::use-pmd-plugin[]
+plugins {
+// end::use-checkstyle-plugin[]
+// end::use-codenarc-plugin[]
+// end::use-findbugs-plugin[]
+// end::use-jdepend-plugin[]
+// end::use-pmd-plugin[]
+    groovy
+// tag::use-checkstyle-plugin[]
+    checkstyle
+// end::use-checkstyle-plugin[]
+// tag::use-codenarc-plugin[]
+    codenarc
+// end::use-codenarc-plugin[]
+// tag::use-findbugs-plugin[]
+    findbugs
+// end::use-findbugs-plugin[]
+// tag::use-jdepend-plugin[]
+    jdepend
+// end::use-jdepend-plugin[]
+// tag::use-pmd-plugin[]
+    pmd
+// tag::use-checkstyle-plugin[]
+// tag::use-codenarc-plugin[]
+// tag::use-findbugs-plugin[]
+// tag::use-jdepend-plugin[]
+}
+// end::use-checkstyle-plugin[]
+// end::use-codenarc-plugin[]
+// end::use-findbugs-plugin[]
+// end::use-jdepend-plugin[]
+// end::use-pmd-plugin[]
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile(localGroovy())
+    testCompile("junit:junit:4.12")
+}
+
+// tag::customize-checkstyle-report[]
+tasks.withType<Checkstyle>() {
+    reports {
+        xml.isEnabled = false
+        html.isEnabled = true
+        html.configure(closureOf<CustomizableHtmlReport> {
```

Comment:
@eriwen this shows a design issue in the API, IMHO. We should be able to configure the stylesheet property the same ay as other properties, without having to cast to CustomizableHtmlReport or to use a groovy closure. 

[`FindBugsXmlReport.xml`](https://github.com/gradle/gradle/blob/e975872ae2cf1d521b73f1107cae772ca09c63cd/subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugsReports.java#L37), for example, is of type `FindBugsXmlReport` (and a few other sub-interfaces use more specific types for the various reports), but most of the properties of the various ReportContainer sub-interfaces use the base report interface for all their reports, which prevents customizing them in a type-safe way, and to discover the API using code-completion in the IDE.

Reply:
There are two issues at play here.
- the Gradle API for configuring reports lack type safe constructs to properly configure the reports
- `html.configure()` is internal (`org.gradle.util` isn't public api)

@jjustinic, could you please open an issue on `gradle/gradle` about the former: the API not providing type safe ways to configure the reports?

In the meantime, here's how I would translate this to be explicit and not use any internal API:

```kotlin
tasks.withType<Checkstyle> {
    reports {
        (html as CustomizableHtmlReport).apply {
            stylesheet = resources.text.fromFile("config/xsl/checkstyle-custom.xsl")
        }
    }
}
```

Reply:
@eskatos Would you consider the later to be a Kotlin DSL issue then? The `html.configure()` method was presented by the IDE as the first suggested method for code completion. I only started with that implementation in the sample because following the IDE suggestion seemed to be what would be expected for most casual users.

Reply:
@jjustinic thank you for opening #6543. The fact that the IDE presented you the internal `.configure()` is because the Gradle API  is leaking internal members.